hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -43,17 +43,45 @@ def __repr__(self):
 
 @attr.s(slots=True)
 class Example(object):
+    """Examples track the hierarchical structure of draws from the byte stream.
+
+    An example usually corresponds to a single draw from a strategy, and
+    includes that strategy's nested draws as children. There are also other
+    kinds of examples, such as leaf examples that correspond to individual
+    low-level blocks, and a single top-level example that spans the entire
+    input.
+
+    Example-tracking allows the shrinker to try "high-level" transformations,
+    such as rearranging or deleting the elements of a list, without having
+    to understand their exact representation in the byte stream.
+    """
+
+    # Depth of this example in the example tree, which is equal to its number
+    # of ancestor examples.
```

Comment:
So the first example drawn would have depth 0?

Reply:
The root example that spans the entire test run has a depth of 0.

Anything that actually gets drawn will end up having a depth of at least 1. I believe the arguments to `@given` end up being packed into a tuple strategy, so the first "user-facing" draw should have a depth of at least 2, though I haven't confirmed this.

(Explicitly listing the depth of the root example seems more useful than what I've currently written here.)